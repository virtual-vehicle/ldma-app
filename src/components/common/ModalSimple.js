
import React from 'react';
import {
  View,
  TouchableOpacity,
  Text,
  StyleSheet,
} from 'react-native';
import Modal from 'react-native-modal';
import { COLORS } from 'ldmaapp/src/constants/colors';

const FIELDS_WIDTH = '88%';
const BUTTON_HEIGHT = 50;

const ModalSimple = (props) => {
  const {
    isVisible,
    onPress,
    headline,
    content,
  } = props;

  return (
    <Modal
      isVisible={isVisible}
      onBackdropPress={() => onPress()}
      style={styles.modalContainer}
    >
      <View style={styles.modalMainSection}>
        {headline && <Text style={styles.modalHeadlineText}>{headline}</Text>}
        {content && <Text style={styles.modalContentText}>{content}</Text>}
        <TouchableOpacity
          style={styles.modalButton}
          onPress={() => onPress()}
        >
          <Text style={styles.modalButtonText}>OK</Text>
        </TouchableOpacity>
      </View>
    </Modal>
  );
};

const styles = StyleSheet.create({
  modalContainer: {
    display: 'flex',
    alignItems: 'center',
  },
  modalMainSection: {
    display: 'flex',
    alignItems: 'center',
    width: '100%',
    backgroundColor: COLORS.WHITE,
    borderRadius: 3,
  },
  modalHeadlineText: {
    color: COLORS.BLACK,
    fontSize: 24,
    textAlign: 'center',
    marginTop: 25,
    fontWeight: '500',
  },
  modalContentText: {
    marginTop: 17,
    fontSize: 18,
    textAlign: 'center',
    color: COLORS.GREY2,
  },
  modalButton: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: COLORS.BLUE,
    width: FIELDS_WIDTH,
    borderRadius: 5,
    height: BUTTON_HEIGHT,
    marginTop: 25,
    marginBottom: 30,
  },
  modalButtonText: {
    color: COLORS.WHITE,
    fontSize: 20,
  },
});

export default ModalSimple;

