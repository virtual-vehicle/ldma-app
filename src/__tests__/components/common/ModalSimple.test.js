import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ModalSimple from '../../../components/common/ModalSimple';

Enzyme.configure({ adapter: new Adapter() });

const headline = 'This is test headline';
const content = 'This is tect content for the modal simple';

describe('ModalSimple component', () => {
  it('renders correctly with all props', () => {
    const onButtonPressMock = jest.fn();
    const wrapper = shallow(<ModalSimple
      isVisible
      onPress={onButtonPressMock}
      headline={headline}
      content={content}
    />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly without content', () => {
    const onButtonPressMock = jest.fn();
    const wrapper = shallow(<ModalSimple
      isVisible
      onPress={onButtonPressMock}
      headline={headline}
    />);
    expect(wrapper).toMatchSnapshot();
  });
  it('close/hide the modal', () => {
    const onButtonPressMock = jest.fn();
    const wrapper = shallow(<ModalSimple
      isVisible
      onPress={onButtonPressMock}
      headline={headline}
      content={content}
    />);
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({ isVisible: false });
    expect(wrapper).toMatchSnapshot();
  });
  it('simulate click OK', () => {
    const onButtonPressMock = jest.fn();
    const wrapper = shallow(<ModalSimple
      isVisible
      onPress={onButtonPressMock}
      headline={headline}
      content={content}
    />);
    expect(wrapper).toMatchSnapshot();
    const buttonElement = wrapper.find('Text').at(2); // step 1, find OK (third text)
    buttonElement.simulate('click'); // step 2, click OK,
    expect(wrapper).toMatchSnapshot();
  });
});
