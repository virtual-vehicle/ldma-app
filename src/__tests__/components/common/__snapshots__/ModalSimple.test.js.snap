// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalSimple component close/hide the modal 1`] = `
<ReactNativeModal
  animationIn="slideInUp"
  animationInTiming={300}
  animationOut="slideOutDown"
  animationOutTiming={300}
  avoidKeyboard={false}
  backdropColor="black"
  backdropOpacity={0.7}
  backdropTransitionInTiming={300}
  backdropTransitionOutTiming={300}
  hideModalContentWhileAnimating={false}
  isVisible={true}
  onBackButtonPress={[Function]}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalShow={[Function]}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
    }
  }
  swipeThreshold={100}
  useNativeDriver={false}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderRadius": 3,
        "display": "flex",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 24,
          "fontWeight": "500",
          "marginTop": 25,
          "textAlign": "center",
        }
      }
    >
      This is test headline
    </Text>
    <Text
      style={
        Object {
          "color": "#4A4A4A",
          "fontSize": 18,
          "marginTop": 17,
          "textAlign": "center",
        }
      }
    >
      This is tect content for the modal simple
    </Text>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#7F7FFF",
          "borderRadius": 5,
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 30,
          "marginTop": 25,
          "width": "88%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFF",
            "fontSize": 20,
          }
        }
      >
        OK
      </Text>
    </TouchableOpacity>
  </View>
</ReactNativeModal>
`;

exports[`ModalSimple component close/hide the modal 2`] = `
<ReactNativeModal
  animationIn="slideInUp"
  animationInTiming={300}
  animationOut="slideOutDown"
  animationOutTiming={300}
  avoidKeyboard={false}
  backdropColor="black"
  backdropOpacity={0.7}
  backdropTransitionInTiming={300}
  backdropTransitionOutTiming={300}
  hideModalContentWhileAnimating={false}
  isVisible={false}
  onBackButtonPress={[Function]}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalShow={[Function]}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
    }
  }
  swipeThreshold={100}
  useNativeDriver={false}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderRadius": 3,
        "display": "flex",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 24,
          "fontWeight": "500",
          "marginTop": 25,
          "textAlign": "center",
        }
      }
    >
      This is test headline
    </Text>
    <Text
      style={
        Object {
          "color": "#4A4A4A",
          "fontSize": 18,
          "marginTop": 17,
          "textAlign": "center",
        }
      }
    >
      This is tect content for the modal simple
    </Text>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#7F7FFF",
          "borderRadius": 5,
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 30,
          "marginTop": 25,
          "width": "88%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFF",
            "fontSize": 20,
          }
        }
      >
        OK
      </Text>
    </TouchableOpacity>
  </View>
</ReactNativeModal>
`;

exports[`ModalSimple component renders correctly with all props 1`] = `
<ReactNativeModal
  animationIn="slideInUp"
  animationInTiming={300}
  animationOut="slideOutDown"
  animationOutTiming={300}
  avoidKeyboard={false}
  backdropColor="black"
  backdropOpacity={0.7}
  backdropTransitionInTiming={300}
  backdropTransitionOutTiming={300}
  hideModalContentWhileAnimating={false}
  isVisible={true}
  onBackButtonPress={[Function]}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalShow={[Function]}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
    }
  }
  swipeThreshold={100}
  useNativeDriver={false}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderRadius": 3,
        "display": "flex",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 24,
          "fontWeight": "500",
          "marginTop": 25,
          "textAlign": "center",
        }
      }
    >
      This is test headline
    </Text>
    <Text
      style={
        Object {
          "color": "#4A4A4A",
          "fontSize": 18,
          "marginTop": 17,
          "textAlign": "center",
        }
      }
    >
      This is tect content for the modal simple
    </Text>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#7F7FFF",
          "borderRadius": 5,
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 30,
          "marginTop": 25,
          "width": "88%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFF",
            "fontSize": 20,
          }
        }
      >
        OK
      </Text>
    </TouchableOpacity>
  </View>
</ReactNativeModal>
`;

exports[`ModalSimple component renders correctly without content 1`] = `
<ReactNativeModal
  animationIn="slideInUp"
  animationInTiming={300}
  animationOut="slideOutDown"
  animationOutTiming={300}
  avoidKeyboard={false}
  backdropColor="black"
  backdropOpacity={0.7}
  backdropTransitionInTiming={300}
  backdropTransitionOutTiming={300}
  hideModalContentWhileAnimating={false}
  isVisible={true}
  onBackButtonPress={[Function]}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalShow={[Function]}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
    }
  }
  swipeThreshold={100}
  useNativeDriver={false}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderRadius": 3,
        "display": "flex",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 24,
          "fontWeight": "500",
          "marginTop": 25,
          "textAlign": "center",
        }
      }
    >
      This is test headline
    </Text>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#7F7FFF",
          "borderRadius": 5,
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 30,
          "marginTop": 25,
          "width": "88%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFF",
            "fontSize": 20,
          }
        }
      >
        OK
      </Text>
    </TouchableOpacity>
  </View>
</ReactNativeModal>
`;

exports[`ModalSimple component simulate click OK 1`] = `
<ReactNativeModal
  animationIn="slideInUp"
  animationInTiming={300}
  animationOut="slideOutDown"
  animationOutTiming={300}
  avoidKeyboard={false}
  backdropColor="black"
  backdropOpacity={0.7}
  backdropTransitionInTiming={300}
  backdropTransitionOutTiming={300}
  hideModalContentWhileAnimating={false}
  isVisible={true}
  onBackButtonPress={[Function]}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalShow={[Function]}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
    }
  }
  swipeThreshold={100}
  useNativeDriver={false}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderRadius": 3,
        "display": "flex",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 24,
          "fontWeight": "500",
          "marginTop": 25,
          "textAlign": "center",
        }
      }
    >
      This is test headline
    </Text>
    <Text
      style={
        Object {
          "color": "#4A4A4A",
          "fontSize": 18,
          "marginTop": 17,
          "textAlign": "center",
        }
      }
    >
      This is tect content for the modal simple
    </Text>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#7F7FFF",
          "borderRadius": 5,
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 30,
          "marginTop": 25,
          "width": "88%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFF",
            "fontSize": 20,
          }
        }
      >
        OK
      </Text>
    </TouchableOpacity>
  </View>
</ReactNativeModal>
`;

exports[`ModalSimple component simulate click OK 2`] = `
<ReactNativeModal
  animationIn="slideInUp"
  animationInTiming={300}
  animationOut="slideOutDown"
  animationOutTiming={300}
  avoidKeyboard={false}
  backdropColor="black"
  backdropOpacity={0.7}
  backdropTransitionInTiming={300}
  backdropTransitionOutTiming={300}
  hideModalContentWhileAnimating={false}
  isVisible={true}
  onBackButtonPress={[Function]}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalShow={[Function]}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
    }
  }
  swipeThreshold={100}
  useNativeDriver={false}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderRadius": 3,
        "display": "flex",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 24,
          "fontWeight": "500",
          "marginTop": 25,
          "textAlign": "center",
        }
      }
    >
      This is test headline
    </Text>
    <Text
      style={
        Object {
          "color": "#4A4A4A",
          "fontSize": 18,
          "marginTop": 17,
          "textAlign": "center",
        }
      }
    >
      This is tect content for the modal simple
    </Text>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#7F7FFF",
          "borderRadius": 5,
          "height": 50,
          "justifyContent": "center",
          "marginBottom": 30,
          "marginTop": 25,
          "width": "88%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFF",
            "fontSize": 20,
          }
        }
      >
        OK
      </Text>
    </TouchableOpacity>
  </View>
</ReactNativeModal>
`;
